src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,109 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int capacity;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        capacity = INITIAL_CAPACITY;
+        threshold = (int) (capacity * LOAD_FACTOR);
+        this.table = (Node<K, V>[]) new Node[capacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkSize();
+        int hashValue = hash(key);
+        int index = findIndex(hashValue);
+        if (table[index] == null) {
+            table[index] = new Node<>(hashValue, key, value, null);
+        } else {
+            Node<K, V> iterator = table[index];
+            while (true) {
+                if (iterator.key == key || Objects.equals(iterator.key, key)) {
+                    iterator.value = value;
+                    return;
+                }
+                if (iterator.next == null) {
+                    break;
+                }
+                iterator = iterator.next;
+            }
+            iterator.next = new Node<>(hashValue, key, value, null);
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = findIndex(hash(key));
+        if (table[index] == null) {
```

Comment:
Are you sure that you need null check here?

Reply:
Not need now